### PR TITLE
👥 remove strip markdown and remark

### PR DESCRIPTION
**Pull request by joshbermanssw**
So we had a PR https://github.com/tinacms/tina.io/issues/2214 to update a dependency `remark`

However i noticed this one dependency was only being used once througout the entire code base for a singular method. Specifically it was beign used to remove markdown syntax from our .md and .mdx files so that we can index them to use in our algolia searches.

In this pr i : 
- removed the strip-markdown and remark dependencies 
- removed indexing on guides as we moved them into docs
- re-wrote the `stripMarkdown` method in our `blog_helper.ts` modelled after https://github.com/zuchka/remove-markdown
- added a test-case to ensure our updated method works for an edge case i found whilst building the new function 

Screenshot 2024-10-09 at 8 55 28 am

**Figure: ✅ Algolia powered Doc search working**

Screenshot 2024-10-09 at 5 38 13 pm

**Figure: ✅ All test cases passing!**

### CODE REVIEW

File: utils/blog_helpers.ts [hunk 1]
```diff
@@ -7,25 +7,44 @@ export function orderPosts(posts) {
...
+      .replace(/!\[([^\]]*)\]\([^\)]+\)/g, '$1')
+      .replace(/#{1,6}\s*(.*)/g, '$1')
+      .replace(/>\s*(.*)/g, '$1')
+      .replace(/(\*\*|__)(.*?)\1/g, '$2')
+      .replace(/(\*|_)(.*?)\1/g, '$2')
+      .replace(/`([^`]+)`/g, '$1')
+      .replace(/^(\n)?\s{0,}#{1,6}\s*( (.+))? +#+$|^(\n)?\s{0,}#{1,6}\s*( (.+))?$/gm, '$1$3$4$6')
+      .replace(/~(.*?)~/g, '$1');
+
+    content = content.replace(/<[^>]*>/g, '');
```
Comment (isaaclombardssw) [inline utils/blog_helpers.ts:33]: AI thinks this is alright.

🤖
The code is not "weird," but it is quite aggressive in sanitizing and formatting text. It aims to strip out various types of formatting, tags, and lists to convert the content to plain text.
🤖

**Review by isaaclombardssw — APPROVED** [review]

File: utils/blog_helpers.ts (hunk 1 above)
Comment (bradystroud) [reply utils/blog_helpers.ts:33]: FYI, we checked out a few packages that achieve the same results and they use very similar code although some more comments could be useful @joshbermanssw  😉

Comment (bradystroud) [reply utils/blog_helpers.ts:33]: Also the tests help with the code being a bit scary.

File: diff.txt
Comment (bradystroud) [inline diff.txt:1]: 🚮

**Review by bradystroud — CHANGES_REQUESTED** [review]

File: utils/blog_helpers.ts (hunk 1 above)
Comment (joshbermanssw) [reply utils/blog_helpers.ts]: added some more comments! Did you give AI the context that the method is used as a markdownStripper?

File: diff.txt
Comment (joshbermanssw) [reply diff.txt:1]: ✅

**Review by bradystroud — APPROVED** [review]
